app/code/Magento/CatalogUrlRewrite/Observer/ProductUrlPathAutogeneratorObserver.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Copyright © 2013-2017 Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\CatalogUrlRewrite\Observer;
+
+use Magento\Catalog\Model\Product;
+use Magento\CatalogUrlRewrite\Model\ProductUrlPathGenerator;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+
+class ProductUrlPathAutogeneratorObserver implements ObserverInterface
+{
+    /** @var ProductUrlPathGenerator */
+    protected $productUrlPathGenerator;
```

Comment:
According to Magento technical guidelines 
http://devdocs.magento.com/guides/v2.1/coding-standards/technical-guidelines/technical-guidelines.html
2.7. All non-public properties and methods SHOULD be private.

Could you change protected visibility to private?